src/GiraffeGenerator.Core/CodeGenNullChecking.fs

```diff
@@ -0,0 +1,212 @@
+﻿module CodeGenNullChecking
+
+open AST
+open ASTExt
+open FSharp.Compiler.SyntaxTree
+open OpenApi
+open OpenApiToAstTypesMatchingAndConversion
+
+type private PropertyPathSegment =
+  | Property of string
+  | NullableValue
+  | CollectionValue of PropertyPathSegment
+  | OptionValue of PropertyPathSegment
+
+let private isNullReference = "isNullReference"
+let private nullReferenceToOption = "nullReferenceToOption"
+let private defaultWithEmptySeq = Option.defaultValueExpr (longIdentExpr "Seq.empty")
+let private checkForUnexpectedNullsName = "checkForUnexpectedNulls"
+let generateNullCheckingHelpers () =
```

Comment:
Just as a refresher, why we need null checking in generated code?
to prevent deseralizing `required string` as `null`?

Shouldn't it be user problem with setting up his serializer (e.g. Newtonsoft.Json) properly?

Reply:
or `required record` as `null`, yes. It may be configured via serializer, yes, but I think that we should check this ourselves because we clearly provide `Option<_>` for optional properties and parameters, so I consider not-null as a part of a contract we provide to the user. E.g. if property/parameter type is not `Option<_>` then we've got some value.
I agree that `null` possibility is an implementation detail of serialization, but I think that the proper implementation of OpenAPI spec should not leave this detail up to user

Reply:
dropped null-checking as unstable and off-topic